infra/README.md

```diff
@@ -0,0 +1,113 @@
+This is a work in progress to use Terraform to manage our cloud
+infrastructure.
+
+Instructions:
+
+- You will need a GCP project.  We assume `gcloud` is configured to
+  point at your project.
+
+- Create a service account for Terraform with Owner role, create a
+  service account key and place it in
+  `$HOME/.hail/terraform_sa_key.json`.
+
+- Enable the the GCP services needed by Hail:
+
+   ```
+   gcloud services enable \
+       compute.googleapis.com \
+       cloudresourcemanager.googleapis.com \
+       servicenetworking.googleapis.com \
+       sqladmin.googleapis.com \
+       container.googleapis.com \
+       serviceusage.googleapis.com
+   ```
+
+- Install terraform.
+
+- Create `$HOME/.hail/global.tfvars` that looks like:
+
+   ```
+   gcp_project = "<gcp-project>"
+   gcp_region = "<gcp-region>"
+   gcp_zone = "<gcp-zone>"
+   domain = "<domain>"
+   ```
+
+- Run `terraform init`.
+
+- Run `terraform apply -var-file="$HOME/.hail/global.tfvars"`.
+  Terraform has created a GKE cluster named `vdc`.  We assume
+  `kubectl` is configured to point at this cluster.
+
+- Go to the Google Cloud console, VPC networks > internal > Private
```

Comment:
I think you might be able to express this in Terraform using `google_compute_network_peering` (https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_network_peering#export_custom_routes), but I haven't looked into it in detail.

Reply:
Yeah, that might work.  I tried it but couldn't get it to work (probably user error).  The docs are written to peer two user-created networks, but we want to peer our internal network with the Google managed Cloud SQL network.  I imagine we'll want to make a pass over all of this to make it more automated.  Let's revisit this then.